services/sidecar/src/simcore_service_sidecar/utils.py

```diff
@@ -109,3 +109,21 @@ def touch_tmpfile(extension=".dat") -> Path:
     """
     with tempfile.NamedTemporaryFile(delete=False, suffix=extension) as file_handler:
         return Path(file_handler.name)
+
+
+def cancel_task(task_name: str) -> None:
+    tasks = asyncio.all_tasks()
+    logger.debug("running tasks: %s", tasks)
+    for task in tasks:
+        if task.get_name() == task_name:
+            logger.warning("canceling task %s....................", task)
+            task.cancel()
```

Comment:
Optimisation proposal in case `task.get_name()` is unique
```suggestion
    tasks_names = {t.get_name(): t for t in tasks}
    if task_name in tasks_names:
        task = tasks_names[task_name]
        logger.warning("canceling task %s....................", task)
        task.cancel()
```

Reply:
thx I added a break there as well...